pkg/minikube/cluster/cluster.go

```diff
@@ -293,16 +296,54 @@ func engineOptions(config MachineConfig) *engine.Options {
 
 func createVirtualboxHost(config MachineConfig) drivers.Driver {
 	d := virtualbox.NewDriver(constants.MachineName, constants.Minipath)
-	d.Boot2DockerURL = config.MinikubeISO
+	d.Boot2DockerURL = config.GetISOCacheLocation()
 	d.Memory = config.Memory
 	d.CPU = config.CPUs
 	d.DiskSize = int(config.DiskSize)
 	return d
 }
 
+func (m *MachineConfig) CacheMinikubeISO() error {
+	// cache the miniube-iso to .minikube dir
+	response, err := http.Get(m.MinikubeISO)
+	if err != nil {
+		return err
+	} else {
+		out, err := os.Create(m.GetISOCacheLocation())
+		if err != nil {
+			return err
+		}
+		defer out.Close()
+		defer response.Body.Close()
+		_, err = io.Copy(out, response.Body)
+		if err != nil {
+			return err
+		}
+	}
+	m.MinikubeISO = "file://" + m.GetISOCacheLocation()
+	return nil
+}
+
+func (m *MachineConfig) GetISOCacheLocation() string {
+	return filepath.Join(constants.Minipath, "cache", "iso", filepath.Base(m.MinikubeISO))
```

Comment:
Although constants.Minipath is also using OS specific separators. Must be a better way to turn file path into file URI.